centreon/www/class/config-generate/abstract/objectJSON.class.php

```diff
@@ -89,8 +89,16 @@ protected function writeFile($dir)
         }
     }
 
-    protected function generateFile($object)
+    protected function generateFile($object, $brokerType = true)
     {
-        $this->content = json_encode(['centreonBroker' => $object], JSON_UNESCAPED_SLASHES|JSON_PRETTY_PRINT);
+        $data = [];
+
+        if ($brokerType) {
+            $data = ['centreonBroker' => $object];
+        } else {
+            $data = $object;
+        }
```

Comment:
Useless init which is overriden.
```suggestion
        if ($brokerType) {
            $data = ['centreonBroker' => $object];
        } else {
            $data = $object;
        }
```